pallets/circuit/src/lib.rs

```diff
@@ -1655,6 +1633,14 @@ impl<T: Config> Pallet<T> {
                 e
             })?;
 
+            if index < side_effects.len() - 1
+                && side_effects[index].reward_asset_id != side_effects[index + 1].reward_asset_id
+            {
+                return Err(
+                    "SFX validate failed - enforce all SFX to have the same reward asset field",
+                )
+            }
+
```

Comment:
can we not do a safe index check by
- fold over each side effect, initial being true
- if asset id != next asset id, set to false
- variable is assets_are_same

This avoids looping each check and can be an easy testable terminal op

Reply:
changed to using .get 
            if let Some(next) = side_effects.get(index + 1) {


not in favor of extra loop since we already enumerate over SFX array